rustls/examples/internal/test_ca.rs

```diff
@@ -138,12 +141,25 @@ fn crl_for_serial(serial_number: SerialNumber) -> CertificateRevocationListParam
     }
 }
 
+fn expired_crl() -> CertificateRevocationListParams {
+    let now = OffsetDateTime::now_utc();
+    CertificateRevocationListParams {
+        this_update: now,
+        next_update: now + Duration::from_secs(60),
+        crl_number: SerialNumber::from(1234),
+        issuing_distribution_point: None,
+        revoked_certs: vec![],
+        key_identifier_method: KeyIdMethod::Sha256,
+    }
+}
+
 // Note: these are ordered such that the data dependencies for issuance are satisfied.
-const ROLES: [Role; 4] = [
+const ROLES: [Role; 5] = [
     Role::TrustAnchor,
     Role::Intermediate,
     Role::EndEntity,
     Role::Client,
+    Role::CrlExpired,
```

Comment:
I think we can probably skip having a `Role` for this, and emit expired CRLs for the non-TrustAnchor roles unconditionally, alongside the exiting `$ROLE..revoked.crl.pem` CRL file as something like `$ROLE.revoked.expired.crl.pem`.

I'm imagining slotting in a new call to your `expired_crl()` fn here-ish: https://github.com/rustls/rustls/blob/ffcc718e23e8f2f8b9aef8035ceb0bd36a7f7f68/rustls/examples/internal/test_ca.rs#L73-L84

Does that make sense? I think it will simplify the `test_ca.rs` diff a good amount.

Reply:
I like that idea 👍